rust/bridge/jni/bin/gen_java_decl.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (C) 2020 Signal Messenger, LLC.
+# SPDX-License-Identifier: AGPL-3.0-only
+#
+
+import os
+import subprocess
+import re
+
+our_abs_dir = os.path.dirname(os.path.realpath(__file__))
+
+cbindgen = subprocess.Popen(['cbindgen'], cwd=os.path.join(our_abs_dir, '..'), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
Nitpick: I suggest passthrough stderr for this.

Reply:
That makes a lot of noise

WARN: Can't find JClass. This usually means that this type was incompatible or not found.
... <497 more of these>


OTOH current version means all failures are silently hidden. How would you feel about capturing stderr into a pipe, but filtering it such that we ignore "This usually means that this type was incompatible or not found." but if any other output goes to stderr we halt immmediately?

Reply:
What happens with cbindgen `[parse]` options `parse_deps = true` and `include = ["jni"]`? Is that sufficient to avoid the incompatible types?

Reply:
If not, yeah, filtering out this particular warning seems sad-but-reasonable.

Reply:
Interesting idea. Doesn't seem to help though, I assume because `JClass` and such are not declared `repr(C)`